generator/sbpg/targets/resources/sbp_tests_template.rs

```diff
@@ -51,15 +44,15 @@ fn test_(((s.suite_name)))()
                 .expect("failed to parse message")
         };
         match &sbp_msg {
-            sbp::messages::SBP::(((t.msg.name)))(msg) => {
-                assert_eq!( msg.get_message_type(), (((t.msg_type))), "Incorrect message type, expected (((t.msg_type))), is {}", msg.get_message_type());
-                let sender_id = msg.get_sender_id().unwrap();
+            sbp::messages::Sbp::(((t.msg.name|lower_acronyms)))(msg) => {
+                assert_eq!( msg.message_type(), (((t.msg_type))), "Incorrect message type, expected (((t.msg_type))), is {}", msg.message_type());
+                let sender_id = msg.sender_id().unwrap();
                 assert_eq!(sender_id, (((t.sbp.sender))), "incorrect sender id, expected (((t.sbp.sender))), is {}", sender_id);
                 ((*- for f in t.fieldskeys *))(((compare_value( (((f))), (((t.fields[f]))) ))))((*- endfor *))
             },
             _ => panic!("Invalid message type! Expected a (((t.msg.name)))"),
         };
-        let frame = sbp_msg.to_frame().unwrap();
+        let frame = sbp::to_vec(&sbp_msg).unwrap();
```

Comment:
nit. What is the idiomatic rule of thumb for using to_asdf vs as_asdf? Is it related to the cost of the action such as a string to slice using "as" vs a vector to string use "to"

Reply:
Yeah pretty much - `as_` should be cheap and not consume the original. `to_` can be more expensive and still does not consume the original. Then `into_` can be either I guess but it should consume the original value. So in this case I went with `to_` because it allocates